src/stringobj.c

```diff
@@ -412,20 +412,14 @@ static Obj FuncREVNEG_STRING(Obj self, Obj val)
 **  character and "first behind last" set to zero
 **
 */
-Obj NEW_STRING ( Int len )
+Obj NEW_STRING(Int len)
 {
-  Obj res;
-  if (len < 0)
-       ErrorQuit(
-           "NEW_STRING: Cannot create string of negative length %d",
-           (Int)len, 0L);
-  res = NewBag( T_STRING, SIZEBAG_STRINGLEN(len)  ); 
-  SET_LEN_STRING(res, len);
-  /* it may be sometimes useful to have trailing zero characters */
-  Char *p = CSTR_STRING(res);
-  p[0] = '\0';
-  p[len] = '\0';
-  return res;
+    if (len < 0)
+        ErrorQuit("NEW_STRING: cannot create string of negative length %d",
+                  (Int)len, 0);
```

Comment:
Actually, thinking about it again, I wonder why we even call `ErrorQuit` here, and not `GAP_ASSERT`, as in other similar functions. And then make it `static inline` and move it into the header file...

Reply:
I agree that I can't think why anyone should be calling such a low-level function with a negative length.